kolibri/plugins/learn/assets/src/actions.js

```diff
@@ -57,21 +60,94 @@ function _collectionState(data) {
 }
 
 
+/*
+* Returns a promise that gets current channel.
+ */
+function _getCurrentChannel() {
+  let currentChannelId = null;
+  return new Promise(
+    (resolve, reject) => {
+      ChannelResource.getCollection({}).fetch()
+        .then((channelList) => {
+          const cookieCurrentChannelId = cookiejs.get('currentChannel');
+          if (channelList.some((channel) => channel.id === cookieCurrentChannelId)) {
+            currentChannelId = cookieCurrentChannelId;
+            resolve(currentChannelId);
+          } else {
+            currentChannelId = channelList[0].id;
+            resolve(currentChannelId);
+          }
+        })
+        .catch((error) => {
+        });
```

Comment:
We shouldn't ever have empty catches - unless we're actually doing something with the error, we should just leave it off.